### PR TITLE
Deprecates use of geometry_msgs/PoseArray for particle cloud in AMCL

### PR DESCRIPTION
**Pull request by abhishek47kashyap**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1679 |
| Primary OS tested on | Ubuntu 20.04.2 LTS |
| Robotic platform tested on | simulated Turtlebot3 in the Gazebo simulator |

---

## Description of contribution in a few bullet points
* Removed publisher `particlecloud_pub_` of type `geometry_msgs/PoseArray`
* Removed `particlecloud_pub_->on_activate()` and `particlecloud_pub_->on_deactivate()`
* Removed a warning message

## Description of documentation updates required from your changes
* [New Particle Filter Messages](https://navigation.ros.org/migration/Eloquent.html?highlight=amcl#new-particle-filter-messages) could be updated to remove mention of `PoseArray`

---

#### For Maintainers: 
- [x] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [x] Check that any new functions have Doxygen added
- [x] Check that any new features have test coverage
- [x] Check that any new plugins is added to the plugins page
- [x] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -873,19 +865,6 @@ AmclNode::publishParticleCloud(const pf_sample_set_t * set)
...
-  auto cloud_msg = std::make_unique<geometry_msgs::msg::PoseArray>();
-  cloud_msg->header.stamp = this->now();
-  cloud_msg->header.frame_id = global_frame_id_;
-  cloud_msg->poses.resize(set->sample_count);
-  for (int i = 0; i < set->sample_count; i++) {
-    cloud_msg->poses[i].position.x = set->samples[i].pose.v[0];
-    cloud_msg->poses[i].position.y = set->samples[i].pose.v[1];
-    cloud_msg->poses[i].position.z = 0;
-    cloud_msg->poses[i].orientation = orientationAroundZAxis(set->samples[i].pose.v[2]);
-    cloud_with_weights_msg->particles[i].pose = (*cloud_msg).poses[i];
```
Comment (SteveMacenski) [inline nav2_amcl/src/amcl_node.cpp:885]: You deleted the population of the particle could poses / weights for the one that's keeping too... I thought you tested this in rviz?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_amcl/src/amcl_node.cpp (hunk 1 above)
Comment (abhishek47kashyap) [reply nav2_amcl/src/amcl_node.cpp:885]: `cloud_msg` is type `PoseArray`, and `cloud_with_weights_msg` is type `ParticleCloud`. I deleted the former.

I did test in rviz, here's a screenshot:
particle_cloud

Comment (SteveMacenski) [reply nav2_amcl/src/amcl_node.cpp:885]: Then where is the particle cloud in that image?

Comment (abhishek47kashyap) [reply nav2_amcl/src/amcl_node.cpp:885]: Dumb mistake, fixed it in 
particle_cloud
f21bed82180dd3765ea5700f873c366c686e2565

**Review by SteveMacenski — APPROVED** [review]
LGTM, waiting on CI!

**Comment by SteveMacenski** [comment]
```
--- src/amcl_node.cpp
+++ src/amcl_node.cpp.uncrustify
@@ -872 +872,2 @@
-    cloud_with_weights_msg->particles[i].pose.orientation = orientationAroundZAxis(set->samples[i].pose.v[2]);
+    cloud_with_weights_msg->particles[i].pose.orientation = orientationAroundZAxis(
+      set->samples[i].pose.v[2]);
@@ -875 +876 @@
```

You have a linting problem

**Comment by SteveMacenski** [comment]
Still linting issues :-)

**Comment by abhishek47kashyap** [comment]
> Still linting issues :-)

Could I may be get the line numbers where my PR fails the lint test?

Suggestions to prevent general lint issues are also welcome. I presume ROS2 uses `cppcheck` and `cpplint`, as [this ROS answer](https://answers.ros.org/question/306194/ros1-and-ros2-linting/?answer=306199#post-id-306199) claims?

**Comment by SteveMacenski** [comment]
If you go into the circle CI job that failed right below, it tells you very directly
```
Diff with 8 lines
--- src/amcl_node.cpp
+++ src/amcl_node.cpp.uncrustify
@@ -873 +873 @@
-            set->samples[i].pose.v[2]);
+      set->samples[i].pose.v[2]);
@@ -876 +876 @@
-  
+

```

Please lint locally, you'd catch this too with `ament_uncrustify`

**Comment by abhishek47kashyap** [comment]
`test_spin_recovery_fake.missing_result`

Originates from
`/opt/overlay_ws/src/navigation2/nav2_system_tests/src/recoveries/spin/test_spin_recovery_node.cpp:79: Failure`

Not sure how to fix this ..

**Comment by SteveMacenski** [comment]
Must have been a flaky run, but the other test build in release mode passed so we can skip it

**Comment by SteveMacenski** [comment]
Thanks for the help!

**Pull request merged**